libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -754,6 +837,342 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
+    connectInfo.userNameLength = pConnectInfo->userNameLength;
+    connectInfo.pPassword = pConnectInfo->pPassword;
+    connectInfo.passwordLength = pConnectInfo->passwordLength;
+    MQTTFixedBuffer_t networkBuffer;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+
+    /*NULL Check willInfo */
+    if( pConnectInfo->pWillInfo != NULL )
+    {
+        MQTTPublishInfo_t willInfo;
+
+        willInfo.retain = pConnectInfo->pWillInfo->retain;
+        willInfo.pTopicName = pConnectInfo->pWillInfo->pTopicName;
+        willInfo.topicNameLength = pConnectInfo->pWillInfo->topicNameLength;
+        willInfo.pPayload = pConnectInfo->pWillInfo->pPayload;
+        willInfo.payloadLength = pConnectInfo->pWillInfo->payloadLength;
+        willInfo.qos = ( MQTTQoS_t ) pConnectInfo->pWillInfo->qos;
+
+        /* Getting Connect packet size using MQTT V4_beta2 API*/
+
+        mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                                &willInfo,
+                                                &remainingLength,
+                                                pPacketSize );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+
+        /*Allocating memory for Connect packet*/
+        networkBuffer.pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.size = *pPacketSize;
+
+        /*Serializing the connect packet using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_SerializeConnect( &connectInfo,
+                                            &willInfo,
+                                            remainingLength,
+                                            &( networkBuffer ) );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+    }
+    else
+    {
+        /* Getting Connect packet size using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_GetConnectPacketSize( &connectInfo,
+                                                NULL,
+                                                &remainingLength,
+                                                pPacketSize );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
+
+        /*Allocating memory for Connect packet*/
+        networkBuffer.pBuffer = IotMqtt_MallocMessage( *pPacketSize );
+        networkBuffer.size = *pPacketSize;
+
+        /*Serializing the connect packet using MQTT V4_beta2 API*/
+        mqttStatus = MQTT_SerializeConnect( &connectInfo,
+                                            NULL,
+                                            remainingLength,
+                                            &( networkBuffer ) );
+
+        /*Converting status code from MQTT v4_beta2 status to MQTT v4_beta 1 status*/
+        status = convertReturnCode( mqttStatus );
```

Comment:
I think the code is repeated for getting packet size and serializing connect. Please refactor to avoid redundant code.